src/js/component.js

```diff
@@ -82,1182 +90,1103 @@ var Component = CoreObject.extend({
-    fn = Lib.bind(this, third);
-
-    // Remove the dispose listener on this component,
-    // which was given the same guid as the event listener
-    this.off('dispose', fn);
-
-    if (first.nodeName) {
-      // Remove the listener
-      Events.off(target, type, fn);
-      // Remove the listener for cleaning the dispose listener
-      Events.off(target, 'dispose', fn);
+    if (typeof first === 'string' || Lib.obj.isArray(first)) {
+      Events.one(this.el_, first, Lib.bind(this, second));
     } else {
-      target.off(type, fn);
-      target.off('dispose', fn);
+      target = first;
+      type = second;
+      fn = Lib.bind(this, third);
+      thisComponent = this;
+
+      newFunc = function(){
+        thisComponent.off(target, type, newFunc);
+        fn.apply(this, arguments);
+      };
+      // Keep the same function ID so we can remove it later
+      newFunc.guid = fn.guid;
+
+      this.on(target, type, newFunc);
     }
-  }
 
-  return this;
-};
+    return this;
+  }
 
-/**
- * Add an event listener to be triggered only once and then removed
- *
- *     myComponent.one('eventName', myFunc);
- *
- * Alternatively you can add a listener to another element or component
- * that will be triggered only once.
- *
- *     myComponent.one(otherElement, 'eventName', myFunc);
- *     myComponent.one(otherComponent, 'eventName', myFunc);
- *
- * @param  {String|vjs.Component}  first   The event type or other component
- * @param  {Function|String}       second  The listener function or event type
- * @param  {Function=}             third   The listener function for other component
- * @return {vjs.Component}
- */
-Component.prototype.one = function(first, second, third) {
-  var target, type, fn, thisComponent, newFunc;
-
-  if (typeof first === 'string' || Lib.obj.isArray(first)) {
-    Events.one(this.el_, first, Lib.bind(this, second));
-  } else {
-    target = first;
-    type = second;
-    fn = Lib.bind(this, third);
-    thisComponent = this;
-
-    newFunc = function(){
-      thisComponent.off(target, type, newFunc);
-      fn.apply(this, arguments);
-    };
-    // Keep the same function ID so we can remove it later
-    newFunc.guid = fn.guid;
+  /**
+   * Trigger an event on an element
+   *
+   *     myComponent.trigger('eventName');
+   *     myComponent.trigger({'type':'eventName'});
+   *
+   * @param  {Event|Object|String} event  A string (the type) or an event object with a type attribute
+   * @return {vjs.Component}       self
+   */
+  trigger(event){
+    Events.trigger(this.el_, event);
+    return this;
+  }
 
-    this.on(target, type, newFunc);
+  /**
+   * Bind a listener to the component's ready state
+   *
+   * Different from event listeners in that if the ready event has already happened
+   * it will trigger the function immediately.
+   *
+   * @param  {Function} fn Ready listener
+   * @return {vjs.Component}
+   */
+  ready(fn){
```

Comment:
I had the thing below, but now I realize it isn't quite what we want, since we do only want to add the function to the queue only if it was given to us.
```js
class Component {
  ready (fn = function() {}) {
    if (this.isReady_) {
      fn.call(this);
    } else {
      this.readyQueue_ = (this.readyQueue_ || []);
      this.readyQueue_.push(fn);
    }
}
```